Simplified/Audiobooks/AudioBookVendors+Extensions.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  AudioBookVendors+Extensions.swift
+//  SimplyE
+//
+//  Created by Vladimir Fedorov on 03.09.2020.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import NYPLAudiobookToolkit
+
+extension AudioBookVendors {
+  
+  /// Vendor tag
+  private var tag: String {
+    "\(FeedbookDRMPublicKeyTag)\(self.rawValue)"
+  }
+  
+  /// UserDefaults key to store certificate date
+  private var validThroughDateKey: String {
+    "\(tag)_validThroughDate"
+  }
+  
+  /// Update vendor's DRM key
+  /// - Parameter completion: Completion
+  func updateDrmCertificate(completion: ((_ error: Error?) -> ())? = nil) {
+    switch self {
+    case .cantook: updateCantookDRMCertificate(completion: completion)
+    }
+  }
+  
+  /// Update Cantook DRM public key
+  ///
+  /// If the key is saved and its saved expiration date is later than today, the function doesn't request a new public key.
+  /// - Parameter completion: Completion
+  private func updateCantookDRMCertificate(completion: ((_ error: Error?) -> ())? = nil) {
+    // Check if we have a valid key
+    if let date = UserDefaults.standard.value(forKey: validThroughDateKey) as? Date, Date() < date {
+      // we have a certificate with a valid date
+      completion?(nil)
+      return
+    }
+    
+    // Fetch a new drmKey
+    DPLAAudiobooks.drmKey { (data, date, error) in
+      if let error = error {
+        if error is DPLAAudiobooks.DPLAError {
+          Log.error(#file, error.localizedDescription)
```

Comment:
I can't quite see it from here, but I wanted to ask if these error situations do end up logging an error on Crashlytics downstream. If so, all good, otherwise we probably should log an explicit event somewhere -- at least on the caller side.

Reply:
That goes to Crashalytics, also added error objects to log as well